Providers/Scripts/2.4x-2.5x/Scripts/nxOMSAuditdPlugin.py

```diff
@@ -8,39 +8,58 @@
 import imp
 import codecs
 import subprocess
+import uuid
+import platform
 
 protocol = imp.load_source('protocol', '../protocol.py')
 nxDSCLog = imp.load_source('nxDSCLog', '../nxDSCLog.py')
 
 LG = nxDSCLog.DSCLog
 
-PLUGIN_CONF = '/etc/audisp/plugins.d/auoms.conf'
-OMSAGENT_CONF = '/etc/opt/microsoft/omsagent/conf/omsagent.d/auoms.conf'
+LOG_FATAL = 'FATAL'
+LOG_ERROR = 'ERROR'
+LOG_INFO = 'INFO'
+
+WORKSPACE_ID_TMPL_STR = "@@WORKSPACE_ID@@"
+AGENT_ETC_ROOT = '/etc/opt/microsoft/omsagent'
+AGENT_VAR_ROOT = '/var/opt/microsoft/omsagent'
+AUOMS_OUTCONF_DIR = '/etc/opt/microsoft/auoms/outconf.d'
+AUOMS_PLUGIN_CONF = '/etc/audisp/plugins.d/auoms.conf'
 MODULE_RESOURCE_DIR = '/opt/microsoft/omsconfig/modules/nxOMSAuditdPlugin/DSCResources/MSFT_nxOMSAuditdPluginResource'
-RESOURCE_OMSAGENT_CONF = MODULE_RESOURCE_DIR + '/conf/auoms.conf'
-RESOURCE_AUDIT_RULES = MODULE_RESOURCE_DIR + '/rules/oms-security-audit.rules'
+RESOURCE_OMSAGENT_CONF = MODULE_RESOURCE_DIR + '/conf/auditd_plugin.conf'
+RESOURCE_AUOMS_CONF = MODULE_RESOURCE_DIR + '/conf/auoms.conf'
+RESOURCE_AUDIT_RULES_V1 = MODULE_RESOURCE_DIR + '/rules/oms-security-audit-v1.rules'
+RESOURCE_AUDIT_RULES_V2 = MODULE_RESOURCE_DIR + '/rules/oms-security-audit-v2.rules'
 
-def Set_Marshall(Ensure):
+def Set_Marshall(WorkspaceId, Ensure):
+    if 'value' in dir(WorkspaceId):
```

Comment:
I don't think you need the if check, but you should encode it as ascii. According to Bhaskar, the MI_String class we use expects ascii, which is why other modules do it. See how nxFileInventory does it for `DestinationPath:` https://github.com/Microsoft/PowerShell-DSC-for-Linux/blob/master/Providers/Scripts/2.4x-2.5x/Scripts/nxFileInventory.py#L51   (it isn't an OMS-specific resource, but Eric Gable wrote it, so I'm sure it's correct.